File: AHome/Ball/Ball/Ball.cs
using System;
using System.Drawing;

namespace Ball
{
    public class Ball
    {
        public Color color;
        public int width;
        public int posX;
        public int posY;

        public int dx;
        public int dy;

        private Random rnd;

        public Ball(int x, int y)
        {
            rnd = new Random();
            color = Color.FromArgb(rnd.Next(0, 255), rnd.Next(0, 255), rnd.Next(0, 255));
            width = rnd.Next(30, 60);
            posX = x;
            posY = y;
            dx = rnd.Next(-10, 10);
            dy = rnd.Next(-10, 10);
        }
    }
}

File: AHome/JS/Ball/ball.js
function Ball(x, y) {
	this.posX = x;
	this.posY = y;
	this.radius = getRandomInt(10, 20);
	this.color = getRandomColor();
	this.dx = getRandomInt(-8, 8);
	this.dy = getRandomInt(-8, 8);

	this.checkHit = function() {
		this.posX += this.dx;
		this.posY += this.dy;

		if ((this.posX - this.radius <= 0) || this.posX + this.radius > canvas.width) {
			this.dx = -this.dx;
		}  
		if ((this.posY - this.radius <= 0) || this.posY + this.radius > canvas.height) {
			this.dy = -this.dy;
		}
	};

	this.drawBall = function(context) {
		context.fillStyle = this.color;
		context.beginPath();
		context.arc(this.posX, this.posY, this.radius, 0, Math.PI * 2, true);
		context.fill();
		context.closePath();
	};
}